src/build_workflow/builder_from_dist.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import urllib.request
+
+from build_workflow.builder import Builder
+from manifests.build_manifest import BuildManifest
+
+
+class BuilderFromDist(Builder):
+    class ManifestGitRepository:
+        def __init__(self, manifest):
+            self.url = manifest.repository
+            self.ref = manifest.ref
+            self.sha = manifest.commit_id
+
+    def checkout(self, work_dir):
+        self.__download_build_manifest()
+
+    def build(self, build_recorder):
+        pass
+
+    def export_artifacts(self, build_recorder):
```

Comment:
`export` doesn't seem to align with the action what about `import` or `download` + "_artifacts"?

Reply:
I thought about this. The "from source" code exports those artifacts, while the "from distribution" code downoads and exports those artifacts. They should really be spit up into `download_artifacts` and `export_artifacts`. I have another PR with more changes along those lines if that's OK?